ctx.go

```diff
@@ -376,54 +411,47 @@ func (c *netHttpContext[B, P]) Params() (P, error) {
 
 	for i := range paramsType.NumField() {
 		field := paramsType.Field(i)
+		fieldValue := paramsValue.Field(i)
 
-		var paramValue string
-		tag := field.Tag.Get("query")
-		if tag != "" {
-			paramValue = c.QueryParam(tag)
-			if paramValue == "" {
-				continue
-			}
-		} else {
-			tag = field.Tag.Get("header")
-			if tag != "" {
-				paramValue = c.Header(tag)
-				if paramValue == "" {
+		// Process query parameters
+		if tag := field.Tag.Get("query"); tag != "" {
+			// Handle slice/array types
+			if field.Type.Kind() == reflect.Slice || field.Type.Kind() == reflect.Array {
```

Comment:
Nit pick: I tend to like switch cases when I see this

Reply:
I approve this message 😁

Reply:
Fixed here